src/main/java/org/gwtproject/tools/cldr/DateTimeFormatInfoProcessor.java

```diff
@@ -35,627 +27,613 @@
+            generateFormat(locale, pw, entry.getValue(), entry.getKey());
+        }
+
+        // write month names
+        generateFullStringList(pw, "month", locale, "months", "1", "2", "3", "4", "5", "6", "7", "8",
+                "9", "10", "11", "12");
+
+        // write quarter names
+        generateStringList(pw, "quarter-wide", null, locale, "quartersFull", "1", "2", "3", "4");
+        generateStringList(pw, "quarter-abbrev", null, locale, "quartersShort", "1", "2", "3", "4");
+
+        // write standard time formats
+        generateArgMethod(pw, "time", locale, "timeFormat");
+        generateStringMethod(pw, "time", locale, "full", "timeFormatFull");
+        generateStringMethod(pw, "time", locale, "long", "timeFormatLong");
+        generateStringMethod(pw, "time", locale, "medium", "timeFormatMedium");
+        generateStringMethod(pw, "time", locale, "short", "timeFormatShort");
+
+        // write weekday names
+        generateFullStringList(pw, "day", locale, "weekdays", DAYS);
+
+        // write weekend boundaries
+        generateDayNumber(pw, locale, "weekendEnd", "weekendEnd");
+        generateDayNumber(pw, locale, "weekendStart", "weekendStart");
+
+        pw.println("}");
+        pw.close();
+    }
+
+    @Override
+    protected void writeOutputFiles() throws IOException {
+        Set<GwtLocale> localesToPrint = localeData.getNonEmptyLocales();
+
+        String factoryPath = "shared/cldr/impl/DateTimeFormatInfo_factory.java";
+        PrintWriter factoryWriter = createOutputFile(factoryPath);
+        printHeader(factoryWriter);
+        factoryWriter.print("package " + "org.gwtproject.i18n.shared.cldr.impl;");
+        // GWT now requires JDK 1.6, so we always generate @Overrides
+        setOverrides(true);
+        factoryWriter.println();
+        factoryWriter.println("// DO NOT EDIT - GENERATED FROM CLDR AND ICU DATA");
+        factoryWriter.println();
+        factoryWriter.println("import org.gwtproject.i18n.shared.cldr.DateTimeFormatInfo;");
+        factoryWriter.println();
+
+        factoryWriter.println("public class DateTimeFormatInfo_factory {");
+        factoryWriter.println(" public static DateTimeFormatInfo create(){");
+        factoryWriter.println();
+
+        List<GwtLocale> sorted = localesToPrint.stream()
+                .sorted(Comparator.comparing(GwtLocale::getAsString)
+                        .reversed())
+                .collect(Collectors.toList());
+
+        for (GwtLocale locale : sorted) {
+
+            String packageName = "org.gwtproject.i18n.shared.cldr.impl";
+
+            String className;
+            if (locale.isDefault()) {
+                className = "DefaultDateTimeFormatInfo";
+            } else {
+                className = "DateTimeFormatInfoImpl" + localeSuffix(locale);
+            }
+
+            factoryWriter.println("   if(System.getProperty(\"locale\").startsWith(\"" + (locale.isDefault() ? "default" : locale.getAsString()) + "\")){");
```

Comment:
what do you think about making this a param, and having it (almost) always be just the constant from System.getProperty? this way, someone _could_ make runtime-locales work, if they wanted, and the compiler wouldn't remove the other locals (and might even optimze to only the ones they wanted)?

Or am i misunderstanding this code and we're already doing that?